File: ruby/bob/bob.rb
class Bob
  def self.hey(line)
    if line.strip.empty?
      "Fine. Be that way!"
    elsif line == line.upcase && line.end_with?("?") && line.match(/[A-Z]+/)
      "Calm down, I know what I'm doing!"
    elsif line.strip.end_with?("?")
      "Sure."
    elsif line.delete('0-9') == line.delete('0-9').upcase && line.match(/[A-Za-z]+/)
      "Whoa, chill out!"
    else
      "Whatever."
    end
  end
end

File: javascript/bob/bob.js
export const hey = (message) => {
  if (message.trim() == '') {
    return 'Fine. Be that way!';
  } else if (message.toUpperCase() == message && message.endsWith('?') && !/^\d+\?/.test(message) && /[a-zA-Z]/.test(message)) {
    return 'Calm down, I know what I\'m doing!';
  } else if (message.toUpperCase() == message && !/^\d+\?/.test(message) && /[a-zA-Z]/.test(message)) {
    return 'Whoa, chill out!';
  } else if (message.trim().endsWith('?')) {
    return 'Sure.';
  } else { return 'Whatever.'; }
};
